server/src/main/java/org/candlepin/pinsetter/core/PinsetterJobListener.java

```diff
@@ -129,6 +130,10 @@ private void updateJob(JobExecutionContext ctx, JobExecutionException exc) {
                 log.error("Job [" + status.getId() + "] failed." , exc);
                 status.setState(JobState.FAILED);
                 status.setResult(exc.getMessage());
+
+                if (exc.getCause() instanceof CandlepinException) {
+                    status.setResultData(((CandlepinException) exc.getCause()).message());
```

Comment:
just curious, why not set the status.result to the cause's method as well? why only status.resultData?